webknossos/webknossos/dataset/dataset.py

```diff
@@ -1002,3 +1118,203 @@ def _initialize_layer_from_properties(self, properties: LayerProperties) -> Laye
+            )
+            assert (
+                dataset_info_update_response.status_code == 200
+            ), dataset_info_update_response
+
+    @property
+    def display_name(self) -> Optional[str]:
+        return self._get_dataset_info().display_name
+
+    @display_name.setter
+    def display_name(self, display_name: Optional[str]) -> None:
+        self._update_dataset_info(display_name=display_name)
+
+    @display_name.deleter
+    def display_name(self) -> None:
+        self.display_name = None
+
+    @property
+    def description(self) -> Optional[str]:
+        return self._get_dataset_info().description
+
+    @description.setter
+    def description(self, description: Optional[str]) -> None:
+        self._update_dataset_info(description=description)
+
+    @description.deleter
+    def description(self) -> None:
+        self.description = None
+
+    @property
+    def tags(self) -> Tuple[str, ...]:
+        return tuple(self._get_dataset_info().tags)
+
+    @tags.setter
+    def tags(self, tags: Sequence[str]) -> None:
+        self._update_dataset_info(tags=list(tags))
+
+    @property
+    def is_public(self) -> bool:
+        return bool(self._get_dataset_info().is_public)
+
+    @is_public.setter
+    def is_public(self, is_public: bool) -> None:
+        self._update_dataset_info(is_public=is_public)
+
+    @property
+    def sharing_token(self) -> str:
+        from webknossos.client._generated.api.default import dataset_sharing_token
+        from webknossos.client.context import _get_generated_client
+
+        with self._context:
+            dataset_sharing_token_response = dataset_sharing_token.sync_detailed(
+                organization_name=self._organization_id,
+                data_set_name=self._dataset_name,
+                client=_get_generated_client(),
+            )
+            assert (
+                dataset_sharing_token_response.status_code == 200
+            ), dataset_sharing_token_response
+            assert dataset_sharing_token_response.parsed is not None
+            return dataset_sharing_token_response.parsed.sharing_token
+
+    @property
+    def allowed_teams(self) -> Tuple["Team", ...]:
+        from webknossos.administration.user import Team
+
+        return tuple(
+            Team(id=i.id, name=i.name, organization_id=i.organization)
+            for i in self._get_dataset_info().allowed_teams
+        )
+
+    @allowed_teams.setter
+    def allowed_teams(self, allowed_teams: Sequence[Union[str, "Team"]]) -> None:
+        from webknossos.administration.user import Team
+        from webknossos.client._generated.api.default import dataset_update_teams
+        from webknossos.client.context import _get_generated_client
+
+        @attr.define
+        class _TeamIdsBody:
+            """Helper class which is deserialized by the generated client via to_dict."""
+
+            teams: List[str]
+
+            def to_dict(self) -> List[str]:
+                return self.teams
```

Comment:
Is there a better way to do this? Why does to_dict return a List?

Reply:
Unfortunately, I don't really think so atm. The origin of this problem is, that the swagger annotation in the backend for the request-payload is not properly typed as a list. I tried to type it this way, but it isn't picked up correctly:
https://github.com/scalableminds/webknossos/blob/fc80c512ee5dd5cba5587ad0e68ccdc6166f6a95/app/controllers/DataSetController.scala#L331-L334
https://webknossos.org/swagger.json

Somehow, this is translated to an empty class `DatasetUpdateTeamsJsonBody` by the client generator. Its instances are deserialized via `to_dict`, as it's assumed that the json body is a json dict. We want to improve the swagger annotations in the future, however, that might involve switching the framework we currently use and is out of scope for this PR.

Reply:
Actually, I just found a much nicer way by modifying `__generate_client.py`, see commit 3c49b40.